Services/Tree/classes/class.ilMaterializedPathTree.php

```diff
@@ -499,7 +499,7 @@ private static function createMaterializedPath($parent, $parentPath)
 	{
 		global $ilDB;
 		$q = ' UPDATE tree
-			SET path = CONCAT(COALESCE(' . $ilDB->quote($parentPath, 'text') . ', \'\'), COALESCE(child, \'\'))
+			SET path = CONCAT(COALESCE(' . $ilDB->quote($parentPath, 'text') . ', \'\'), COALESCE(cast(child as varchar), \'\'))
```

Comment:
@smeyer-ilias could you please review these changes in the Tree-Classes? Thank you very much

Reply:
Have to revert my approval.
The part "cast(child as varchar)" is not standard SQL.

Reply:
CAST(FIELD AS TYPE) is part of SQL92...
The Problem is, that postgres needs explicit casts (concat with different types)

Reply:
At least this will fail with Oracle I think, there it should be a `VARCHAR2`. This Query is tricky... I'm currently thinking of abstracting `CAST`

Reply:
ok, now whe have two options:
- find a sql-statement that will work on all databases
- abstract the method "`cast`"

for the first option this should work:
```SQL
COALESCE(TRIM(CAST(child AS CHAR(2000))), '')
```

for the second option: 
i can implement a new method in ilDBInterface similar to groupConcat in PR #866

what are your preferences?

Reply:
@lbruening Yes, your code is compliant to SQL92, but the data types are handled differently (MySQL, Oracle). 

MySQL: 
cast(something as varchar) is not supported
cast(something as char) or cast(something as char(4000)) is supported

Oracle [1]:
select cast(banner as char(4000)) from v$version; => size to big for datatype
select cast(banner as char(1000)) from v$version; => ok

[1] https://www.tutorialspoint.com/oracle_terminal_online.php

Reply:
...and in Postgres, a cast(something as char(4000)) will return a string of size 4000 (padded with spaces.... damned!)

@chfsx 
ok, i think, i will create an abstraction of the cast-method in ilDBInterface... is this ok for you?

Reply:
@lbruening absolutely, and thank you very much for all your time you invest in this!